src/dftbp/dftb/mdftb.F90

```diff
@@ -0,0 +1,1575 @@
+    real(dp), allocatable :: PyYDxxyy(:)
+    real(dp), allocatable :: PzZDzz(:)
+    real(dp), allocatable :: SXXS(:)
+    real(dp), allocatable :: PxXXPx(:)
+    real(dp), allocatable :: PyXXPy(:)
+    real(dp), allocatable :: SXXDxxyy(:)
+    real(dp), allocatable :: SXXDzz(:)
+    real(dp), allocatable :: SYYDxxyy(:)
+    real(dp), allocatable :: SZZDzz(:)
+    real(dp), allocatable :: DxyXXDxy(:)
+    real(dp), allocatable :: DyzXXDyz(:)
+    real(dp), allocatable :: DxxyyXXDzz(:)
+    real(dp), allocatable :: DzzXXDzz(:)
+    real(dp), allocatable :: DxxyyYYDzz(:)
+    real(dp), allocatable :: DzzZZDzz(:)
+    real(dp), allocatable :: DxzXZDzz(:)
+    real(dp), allocatable :: DyzYZDxxyy(:)
+  end type TMdftbAtomicIntegrals
+
+  !> Input for the MultiExpan module
+  type TMdftbInp
+
+    !> Orbital information
+    integer :: nOrb, nSpin, nSpecies
+    type(TOrbitals), pointer :: orb
+
+    !> Hubbard U values for atoms
+    real(dp), allocatable :: hubbu(:)
+
+    !> Species of atoms
+    integer, allocatable :: species(:)
+
+    !> One-center atomic integral data
+    type(TMdftbAtomicIntegrals), allocatable :: mdftbAtomicIntegrals
+
+  end type TMdftbInp
+
+  !> Internal status of TMultiExpan.
+  type TMdftb
+    integer :: nAtoms, nSpecies, mOrb, nOrb, nSpin
+
+    !> Hubbard U values for atoms
+    real(dp), allocatable :: hubbu(:)
+
+    !> Species of atoms
+    integer, allocatable :: species(:), nOrbSpecies(:)
+    integer, allocatable :: onDQOCharges(:,:)
+
+    !> coordinates of the atom
+    real(dp), allocatable :: coords(:,:)
+
+    real(dp), allocatable :: atomicDIntgrl(:,:,:,:)
+    real(dp), allocatable :: atomicQIntgrl(:,:,:,:,:)
+
+    !> Mulliken charge per atom
+    real(dp), allocatable :: deltaMAtom(:)
+    !> Dipole charge per atom
+    real(dp), allocatable :: deltaDAtom(:,:)
+    !> Quadrupole charge per atom
+    real(dp), allocatable :: deltaQAtom(:,:,:)
+
+    !> evaluated for the E, Atom1, Atom2 at each geometry step
+    real(dp), allocatable :: f10AB(:,:,:)
+    real(dp), allocatable :: f20AB(:,:,:,:)
+    real(dp), allocatable :: f30AB(:,:,:,:,:)
+    real(dp), allocatable :: f40AB(:,:,:,:,:,:)
+    !> evaluated for the gradient, Atom1, Atom2 at each geometry step
+    real(dp), allocatable :: f50AB(:,:,:,:,:,:,:)
+
+    !> add for the H and the E
+    real(dp), allocatable :: pot10x1Atom(:)
+    real(dp), allocatable :: pot20x2Atom(:)
+    real(dp), allocatable :: pot10x0Atom(:,:)
+    real(dp), allocatable :: pot11x1Atom(:,:)
+    real(dp), allocatable :: pot21x2Atom(:,:)
+    real(dp), allocatable :: pot20x0Atom(:,:,:)
+    real(dp), allocatable :: pot21x1Atom(:,:,:)
+    real(dp), allocatable :: pot22x2Atom(:,:,:)
+    !> add for the gradient
+    real(dp), allocatable :: pot30x0Atom(:,:,:,:)
+    real(dp), allocatable :: pot31x1Atom(:,:,:,:)
+    real(dp), allocatable :: pot32x2Atom(:,:,:,:)
+
+    !> total energy
+    real(dp) :: energyTT, energyMD, energyDD, energyMQ, energyDQ, energyQQ
```

Comment:
Also here, let's give default initialization values (to avoid indeterministic behavior due to uninitialized variables)